tests/foreman/virtwho/utils.py

```
@@ -0,0 +1,185 @@
+    runcmd('subscription-manager unregister', system)
+    runcmd('subscription-manager clean', system)
+    runcmd('rpm -qa | grep katello-ca-consumer | xargs rpm -e |sort', system)
+    runcmd(
+        'rpm -ihv http://{}/pub/katello-ca-consumer-latest.noarch.rpm'
+        .format(settings.server.hostname), system
+    )
+    cmd = 'subscription-manager register --org={} '.format(org)
+    if activation_key is not None:
+        cmd += '--activationkey={}'.format(activation_key)
+    else:
+        cmd += '--username={} --password={}'.format(
+            settings.server.admin_username,
+            settings.server.admin_password)
+    ret, stdout = runcmd(cmd, system)
+    if ret == 0 or "system has been registered" in stdout:
+        return True
+    else:
+        raise VirtWhoError(
+            'Failed to register system: {}'.format(system)
+        )
+
+
+def virtwho_cleanup():
+    """Before running test cases, need to clean the environment.
+    Do the following:
+    1. stop virt-who service.
+    2. kill all the virt-who pid
+    3. clean rhsm.log message, make sure there is no old message exist.
+    4. clean all the configure files in /etc/virt-who.d/
+    """
+    runcmd("systemctl stop virt-who")
+    runcmd(
+        "ps -ef | grep virt-who -i | grep -v grep |"
+        "awk '{print $2}' | xargs -I {} kill -9 {}"
+    )
+    runcmd("rm -f /var/run/virt-who.pid")
+    runcmd("rm -rf /var/log/rhsm/*")
+    runcmd("rm -rf /etc/virt-who.d/*")
+
+
+def get_virtwho_status():
+    """Return the status of virt-who service, it will help us to know
+    the virt-who configure file is deployed or not.
+    """
+    ret, stdout = runcmd('systemctl status virt-who')
+    running_stauts = ['is running', 'Active: active (running)']
+    stopped_status = ['is stopped', 'Active: inactive (dead)']
+    if ret == 0 and any(key in stdout for key in running_stauts):
+        return 'running'
+    elif ret == 0 and any(key in stdout for key in stopped_status):
```

Instead of checking two times for the same condition, you could check for it first and fail early:
```python
    running_stauts = ['is running', 'Active: active (running)']
    stopped_status = ['is stopped', 'Active: inactive (dead)']
    ret, stdout = runcmd('systemctl status virt-who')
    if ret != 0:
        return 'undefined'
    if any(key in stdout for key in running_stauts):
        return 'running'
    elif any(key in stdout for key in stopped_status):
        return 'stopped'
    else:
        return 'undefined'
```

---

Make sense to me